app/scripts/models/auth_brokers/oauth.js

```diff
@@ -207,7 +210,14 @@ define(function (require, exports, module) {
 
     afterSignInConfirmationPoll (account) {
       return this.finishOAuthSignInFlow(account)
-        .then(() => proto.afterSignInConfirmationPoll.call(this, account));
+        .then((serviceRedirect) => {
+          if (this._isChromeAndroid) {
+            return new NavigateBehavior('signin_confirmed', {serviceRedirect});
```

Comment:
The base `oauth` broker is supposed to be generic, I think this logic should probably be moved into `oauth-redirect` or as @rfk suggests, into its own broker.